libs/blocks/merch-card/merch-card.js

```diff
@@ -1,184 +1,186 @@
-  }
+  const badgeText = ribbonMetadata[1].innerText;
   ribbonWrapper.remove();
+  return { badgeBackgroundColor, badgeColor, badgeText };
 };
 
-const decorateIcon = (el, icons, cardType) => {
-  if (!icons) return;
-  const inner = el.querySelector(`.consonant-${cardType}-inner`);
-  const iconWrapper = createTag('div', { class: `consonant-${cardType}-iconWrapper` });
-  icons?.forEach((icon) => {
-    const url = icon.querySelector('img').src;
-    const iconDiv = createTag('div', { class: 'consonant-MerchCard-ProductIcon', style: `background-image: url(${url})` });
-    iconWrapper.appendChild(iconDiv);
-    icon.parentNode?.remove();
-  });
-  inner.prepend(iconWrapper);
+const getActionMenuContent = (el, ribbonMetadata) => {
+  const index = (ribbonMetadata !== null) ? 1 : 0;
+  const expectedChildren = ribbonMetadata !== null ? 3 : 2;
+  if (el.childElementCount !== expectedChildren) {
+    return null;
+  }
+  const actionMenuContentWrapper = el.children[index];
+  const actionMenuContent = actionMenuContentWrapper.children[0];
+  actionMenuContentWrapper.remove();
+  return actionMenuContent;
 };
 
+function getMerchCardRows(rows, ribbonMetadata, cardType, actionMenuContent) {
+  if (cardType === 'catalog') {
+    const index = ribbonMetadata === null ? 0 : 1;
+    return actionMenuContent !== null ? rows[index + 1] : rows[index];
+  }
+  return rows[ribbonMetadata === null ? 0 : 1];
+}
+
 const init = (el) => {
-  loadStyle(`${base}/deps/caas.css`);
-  const section = el.closest('.section');
-  section.classList.add('milo-card-section');
-  const images = el.querySelectorAll('picture');
-  let image;
-  const icons = [];
-  const rows = el.querySelectorAll(':scope > *');
-  const ribbonMetadata = rows[0].children?.length === 2 ? rows[0].children : null;
-  const row = rows[ribbonMetadata === null ? 0 : 1];
-  const altCta = rows[rows.length - 1].children?.length === 2
-    ? rows[rows.length - 1].children : null;
-  const allPElems = row.querySelectorAll('p');
-  const ctas = allPElems[allPElems.length - 1];
-  const styles = [...el.classList];
-  const cardType = getPodType(styles);
-  const merchCard = el;
-  decorateBlockHrs(el);
-  images.forEach((img) => {
-    const imgNode = img.querySelector('img');
-    const { width, height } = imgNode;
-    const isSquare = Math.abs(width - height) <= 10;
-    if (img) {
-      if (isSquare) {
-        icons.push(img);
-      } else {
-        image = img;
+  let section = el.closest('.section');
+  let upClass = getUpFromSectionMetadata(section);
+  if (upClass) {
+    initLegacyMerchCard(el);
+  } else {
+    if (section.parentElement.classList.contains('fragment')) {
+      section.style.display = 'contents';
+      const fragment = section.parentElement;
+      fragment.style.display = 'contents';
+      const fragmentParent = fragment.parentElement;
+      fragmentParent.style.display = 'contents';
+      section = fragmentParent.parentElement;
```

Comment:
this is to support fragments, right? 
we could see later if inline fragments can be used instead, but i think it can be a future optimization since we are in a bit of a hurry with this PR.
in case you want already to try it out its just about adding '#_inline' in the end of referenced fragment.

Reply:
Yes, it's for supporting fragments. I will git it a try.